pkg/api/v1alpha1/tinkerbellmachineconfig_webhook_test.go

```diff
@@ -0,0 +1,79 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestTinkerbellMachineConfig_ValidateCreateSuccess(t *testing.T) {
+	machineConfig := createTinkerbellMachineConfig()
+
+	g := NewWithT(t)
+	g.Expect(machineConfig.ValidateCreate()).To(Succeed())
+}
+
+func TestTinkerbellMachineConfig_ValidateCreateFail(t *testing.T) {
+	machineConfig := createTinkerbellMachineConfig(func(mc *TinkerbellMachineConfig) {
+		mc.Spec.HardwareSelector = nil
+	})
+
+	g := NewWithT(t)
+	g.Expect(machineConfig.ValidateCreate()).NotTo(Succeed())
+}
+
+func TestTinkerbellMachineConfig_ValidateUpdateSucceed(t *testing.T) {
+	machineConfigOld := createTinkerbellMachineConfig()
+	machineConfigNew := machineConfigOld.DeepCopy()
+
+	g := NewWithT(t)
+	g.Expect(machineConfigNew.ValidateUpdate(machineConfigOld)).To(Succeed())
+}
+
+func TestTinkerbellMachineConfig_ValidateUpdateFailOSFamily(t *testing.T) {
+	machineConfigOld := createTinkerbellMachineConfig()
+	machineConfigNew := createTinkerbellMachineConfig(func(mc *TinkerbellMachineConfig) {
+		mc.Spec.OSFamily = Bottlerocket
+	})
+
+	g := NewWithT(t)
+	g.Expect(machineConfigNew.ValidateUpdate(machineConfigOld)).NotTo(Succeed())
```

Comment:
Instead of checking NotTo(Succeed), let's validate the right failure message is what we are checking for. This would help confirm that the failure is from OS Family being immutable.